### ISSUE TITLE
New Navbar!

### ISSUE THREAD

**Issue by dallinbjohnson** [papercss/papercss#144]
I would like to contribute a Navbar for Papercss! @rhyneav
![screen shot 2018-04-01 at 12 20 35 pm](https://user-images.githubusercontent.com/25631483/38176210-5186006a-35a8-11e8-82c3-95df40ca826b.png)
![screen shot 2018-04-01 at 12 20 58 pm](https://user-images.githubusercontent.com/25631483/38176220-73170ab2-35a8-11e8-84f9-6467753bf21d.png)
![screen shot 2018-04-01 at 12 22 08 pm](https://user-images.githubusercontent.com/25631483/38176227-8714ddfa-35a8-11e8-9b11-57ec7a9a28af.png)
![screen shot 2018-04-01 at 12 22 55 pm](https://user-images.githubusercontent.com/25631483/38176236-a4d80b00-35a8-11e8-8850-f77bd865a3f3.png)

**Comment by rhyneav** [comment]
Awesome! Love that it's responsive. Is that entirely CSS? 

To create a pull request, fork the PaperCSS repo to your account. Then follow [the instructions here](https://github.com/papercss/papercss#contributing) to create a new branch with your navbar feature in it. Be sure to check out #39 as well for comments on this feature. It might be a good place to start!

For adding the styles, I'd recommend creating a `_navbar.scss` file in `src/components`. For adding the documentation and displaying it on the site, you can create a `navbar.md` file in `docs/content/docs/components/`. Feel free to use the existing files as examples to help you out!

**Comment by dallinbjohnson** [comment]
It is mostly css! but to make it responsive I had to write a little JS to make it work how it has been set up. @rhyneav

**Comment by koester** [comment]
I like the concept, but why don't you change the HTML Markup to use an input as a trigger for the burger icon?

I guess you only added JS for an event listener? IMHO you can work around that with just HTML and CSS.

**Comment by rhyneav** [comment]
Good call, @koester! I believe that's what was done in #39 as well.

The HTML snippet:
```html
<nav>
      <label for="menu-toggle">&#9776;</label>
      <input type="checkbox" id="menu-toggle" role="button"></input>
      <ul>
        <li class="active"><a href="#">Home</a></li>
        <li><a href="#">About</a></li>
        <li><a href="#">Portfolio</a></li>
        <li><a href="#">News</a></li>
        <li><a href="#">Contact</a></li>
      </ul>
</nav>
```

**Comment by dallinbjohnson** [comment]
This is the HTML That I am using for the navbar. The js I am using just unchecks the checkbox when it's a small screen and checks it again when its big.  Thanks for the links that yall have given me to check out.  I have not figured out how to have the navbar hide when it gets small and shows when it gets big. 

```<nav class="fixed">
  <div>
    <h3 class="nav-brand"><a href="#">Get PaperCSS</a></h3>
  </div>
  <div>
    <div class="collapsible">
      <input id="collapsible1" type="checkbox" name="collapsible" checked>
      <button class="btn-small btn-mobile">
        <label for="collapsible1">
          <!-- <i class="fas fa-3x fa-bars"></i> -->
          <div class="bar1"></div>
          <div class="bar2"></div>
          <div class="bar3"></div>
        </label>
      </button>
      <div class="collapsible-body">
        <ul class="inline">
          <li><a href="#">Documentation</a></li>
          <li><a href="#">About</a></li>
          <li><a href="#">Github</a></li>
        </ul>
      </div>
    </div>
  </div>
</nav>```

**Comment by dallinbjohnson** [comment]
This is the all the js that I am using.

```javascript
$(window).on("load, resize", function() {
	if ($(window).width() <= 852){	
		$("nav input[type=checkbox]").removeAttr('checked');
	}
	else {
		$("nav input[type=checkbox]").replaceWith("<input id='collapsible1' type='checkbox' name='collapsible' checked>");
	}
});
```

**Comment by koester** [comment]
I think the main problem lies in your HTML Markup. Check the source on this site: https://www.rockahula.de or #39 

You dont have to check or uncheck the checkbox based on screen width. Think about it this way: The input is only used as a trigger to show the menu when its hidden and close it when its open. On laptop and desktop devices there is no need to trigger anything, so you just hide the checkbox and show the menu as usual. When you go down to a tablet width you just switch up the css with some media querys so that now the checkbox is visible and the navigation is hidden. 

Make changes to the css accordingly so the navigation looks good again on mobile devices and just switch it on / off witch the input and some css magic.

The link I shared illustrates this pretty good (just dont judge the design please :laughing: ). PaperCSS provides a nice mixing to work with media querys. See: https://github.com/papercss/papercss/blob/c395349afd9e0fed09ec28b133c5b4e2b552a2e7/src/core/_config.scss#L124-L138 
To get an idea of how to use the mixin you can check the various components, for example the _button component: https://github.com/papercss/papercss/blob/c395349afd9e0fed09ec28b133c5b4e2b552a2e7/src/components/_buttons.scss#L17-L22

If you could provide a CodePen or JSFiddle with your navbar demo I can provide some additional help, if needed.

**Comment by dallinbjohnson** [comment]
CodePen: https://codepen.io/Diddle-dj/pen/QmxpGG

**Comment by dallinbjohnson** [comment]
never mind I figured out the problem!  Thanks, ya'll!!

**Comment by rhyneav** [comment]
Wonderful explanation @koester!

**Comment by rhyneav** [comment]
Merged in #145!
